winfsp/src/filesystem/context.rs

```diff
@@ -196,6 +196,14 @@ pub trait FileSystemContext<const DIR_INFO_SIZE: usize = MAX_PATH>: Sized {
         Err(STATUS_INVALID_DEVICE_REQUEST.into())
     }
 
+    fn can_delete<P: AsRef<WCStr>>(
```

Comment:
`CanDelete` is functionally identical to `SetDelete(..., true)`. Because its impossible to partially implement a trait, there's no way to tell which of `CanDelete` or `SetDelete` is implemented. 

> NOTE: If both CanDelete and SetDelete are defined, SetDelete takes precedence. However most file systems need only implement the CanDelete operation.

Since there's no way to tell which one is implemented, `Interface` sets the function pointer of `SetDelete` to non-null so [WinFSP will always call `SetDelete`](https://github.com/winfsp/winfsp/blob/0a91292e0502d6629f9a968a168c6e89eea69ea1/src/dll/fsop.c#L1157), regardless of whether or not `CanDelete` is overridden by the trait impl. 

That's also the reason why `create` and `overwrite` are actually the Ex versions, and `GetDirectoryByName` is implemented so weirdly with the `DirectoryResolveStrategy` trait, because otherwise it will always call `GetDirectoryByName` (as it's non-null), albeit when `PassQueryDirectoryFileName` is set.

Reply:
Sounds good! I removed them all